pkg/resources/grant_privileges_to_share.go

```diff
@@ -0,0 +1,474 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"slices"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var grantPrivilegesToShareGrantExactlyOneOfValidation = []string{
+	"database_name",
+	"schema_name",
+	// "function_name",
+	"table_name",
+	"all_tables_in_schema",
+	"tag_name",
+	"view_name",
+}
+
+var grantPrivilegesToShareSchema = map[string]*schema.Schema{
+	"share_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "The fully qualified name of the share on which privileges will be granted.",
```

Comment:
validation?

Reply:
True, I had it at the beginning but removed it when changing share_name for some reason. Added again.